### ISSUE TITLE
Error with custom theme submenus when using pygame_menu.baseimage.BaseImage 

### ISSUE THREAD

**Issue by Uskompuf** [ppizarror/pygame-menu#198]
When I am making a custom theme from an existing theme and use `background.color` with `pygame_menu.baseimage.BaseImage` to set a custom image as menu background the image appears fine but if I go into a submenu then back into the main menu the submenu bar does not disappear.

![image](https://user-images.githubusercontent.com/22492406/82113532-1289cd00-979a-11ea-88d4-59cd47587def.png)

![image](https://user-images.githubusercontent.com/22492406/82113539-1c133500-979a-11ea-84d9-cb2e16fc533c.png)

![image](https://user-images.githubusercontent.com/22492406/82113542-23d2d980-979a-11ea-9f53-660ae9f52c02.png)

**Comment by Uskompuf** [comment]
@anxuae any ideas? Is further code snippets required to diagnose the issue?

**Comment by Uskompuf** [comment]
I have identified the issue to be the use of a transparent png image. I believe this is an issue with the library

**Comment by ppizarror** [comment]
I'll look further, this issue has not been resolved. The image you used is a transparent png?

**Comment by Uskompuf** [comment]
Yes, also is there a way to change a button widget title(text displayed)? `menu.get_current().get_widget("id", recursive=True).set_title("new title")` doesn't seem to work

**Comment by ppizarror** [comment]
I see two problems from the example you provided:

![image](https://user-images.githubusercontent.com/12925256/82176499-eb6bf200-98a4-11ea-9558-7682ce978804.png)

I think this is a problem of the display not being updated

**Comment by ppizarror** [comment]
I can't reproduce this issue. What version of python are you using? Can you prove some minimal code to test this issue? Tested in py2.7 and py 3.7. Also tested a transparent png

![83be2fa5-4b7f-44ba-a8cb-156fc587b7f2](https://user-images.githubusercontent.com/12925256/82177531-b7460080-98a7-11ea-8c6f-f1831ce04acb.gif)

**Comment by Uskompuf** [comment]
Here is some minimal code:

```
import pygame
import pygame_menu

pygame.init()
surface = pygame.display.set_mode((1280, 720))

myimage = pygame_menu.baseimage.BaseImage(
    image_path="test.png",
    drawing_mode=pygame_menu.baseimage.IMAGE_MODE_SIMPLE
)

mytheme = pygame_menu.themes.THEME_DARK.copy()
mytheme.title_bar_style = pygame_menu.widgets.MENUBAR_STYLE_UNDERLINE_TITLE
mytheme.widget_alignment = pygame_menu.locals.ALIGN_LEFT
mytheme.widget_margin = [50, 20]
mytheme.widget_font_size = 40
mytheme.title_font_size = 75
mytheme.background_color = myimage

main_menu = pygame_menu.Menu(720, 1280, 'Casino Game', theme=mytheme)

info_menu = pygame_menu.Menu(720, 1280, 'Game Information', theme=pygame_menu.themes.THEME_DARK)

main_menu.add_button('Game Information', info_menu)

main_menu.mainloop(surface)
```

Here is the image used for testing:
![test](https://user-images.githubusercontent.com/22492406/82177826-ff076f00-991d-11ea-842e-e43e6fbce5db.png)

Here is the result:
![image](https://user-images.githubusercontent.com/22492406/82177861-134b6c00-991e-11ea-9022-694978c781b5.png)

**Comment by ppizarror** [comment]
> Yes, also is there a way to change the text displayed on a button widget with a command? `menu.get_current().get_widget("id", recursive=True).set_title("new title")` doesn't seem to work

set_title does not exists on button api
https://pygame-menu.readthedocs.io/en/stable/_source/widgets_button.html

Where did you find that method? Anyway, I've added set_title() to button class. (https://github.com/ppizarror/pygame-menu/commit/c00b05f9e09910dbf7956bf0b54d373baaeb631e)

**Comment by ppizarror** [comment]
> Here is some minimal code:
> 
> ```
> import pygame
> import pygame_menu
> 
> pygame.init()
> surface = pygame.display.set_mode((1280, 720))
> 
> myimage = pygame_menu.baseimage.BaseImage(
>     image_path="test.png",
>     drawing_mode=pygame_menu.baseimage.IMAGE_MODE_SIMPLE
> )
> 
> mytheme = pygame_menu.themes.THEME_DARK.copy()
> mytheme.title_bar_style = pygame_menu.widgets.MENUBAR_STYLE_UNDERLINE_TITLE
> mytheme.widget_alignment = pygame_menu.locals.ALIGN_LEFT
> mytheme.widget_margin = [50, 20]
> mytheme.widget_font_size = 40
> mytheme.title_font_size = 75
> mytheme.background_color = myimage
> 
> main_menu = pygame_menu.Menu(720, 1280, 'Casino Game', theme=mytheme)
> 
> info_menu = pygame_menu.Menu(720, 1280, 'Game Information', theme=pygame_menu.themes.THEME_DARK)
> 
> main_menu.add_button('Game Information', info_menu)
> 
> main_menu.mainloop(surface)
> ```
> 
> Here is the image used for testing:
> ![test](https://user-images.githubusercontent.com/22492406/82177826-ff076f00-991d-11ea-842e-e43e6fbce5db.png)
> 
> Here is the result:
> ![image](https://user-images.githubusercontent.com/22492406/82177861-134b6c00-991e-11ea-9022-694978c781b5.png)

Now I see, `bgfun=None` in mainloop call, so the surface is not being cleared through each execution of the loop. That's for sure a problem.
For avoiding this issue, I'll create a new parameter on themes as "surface_clear_color". By default that parameter will be black.

The order of the drawing will be the following:

theme.surface_clear_color() # This fills the entire surface
menu.bgfun() # Executes the user background method if provided (If None this is not executed)
draw menu()

**Comment by Uskompuf** [comment]
> Anyway, I've added set_title() to button class.

Can you also add it for label class please?

**Comment by Uskompuf** [comment]
Sounds good

**Comment by ppizarror** [comment]
The background issue has been solved.
![755abf36-dee0-4046-a9a5-3cee9314faee](https://user-images.githubusercontent.com/12925256/82179684-e9a62c80-98ac-11ea-86ff-da91867f317f.gif)

**Comment by Uskompuf** [comment]
> The background issue has been solved.

That's great to hear!

**Comment by ppizarror** [comment]
> > Anyway, I've added set_title() to button class.
> 
> Can you also add it for label class please?

Seems good to be added to all proper classes.

**Comment by Uskompuf** [comment]
> Seems good to be added to all proper classes.

Yes that would be optimal

**Comment by ppizarror** [comment]
#200 implements all changes. Will publish it in a couples of hours. Let me know if there's something to improve in this thread.

**Comment by Uskompuf** [comment]
Will you release a new pip version with these updates or should I just install from github?

**Comment by ppizarror** [comment]
Yes, all new versions are added to pip. But I can't upload the version as there's more bug fixes #199 and #172
